modules/swagger-codegen/src/main/resources/kotlin-client/api.mustache

```diff
@@ -5,12 +5,13 @@ package {{apiPackage}}
 {{/imports}}
 
 import {{packageName}}.infrastructure.*
+import okhttp3.OkHttpClient
 {{#threetenbp}}
 import org.threeten.bp.LocalDateTime
 {{/threetenbp}}
 
 {{#operations}}
-class {{classname}}(basePath: kotlin.String = "{{{basePath}}}") : ApiClient(basePath) {
+class {{classname}}(basePath: kotlin.String = "{{{basePath}}}", httpClient: OkHttpClient = OkHttpClient()) : ApiClient(basePath, httpClient) {
```

Comment:
have you considered putting the httpClient as the first arg? That would make it so that you can override the httpClient without having to supply the url

Reply:
I also wonder if it would be better to user OkHttpClient.Builder().build()? I'm not super versed in OkHttp internals so not sure if there's a real difference